File: backend/board-service/src/main/java/com/example/boardservice/cache/RedisConstants.java
package com.example.boardservice.cache;

import java.time.Duration;

public final class RedisConstants {
    public static final String USER_PROFILE_KEY = "user:profile:%d";
    public static final Duration USER_PROFILE_TTL = Duration.ofHours(1);

    public static final String USER_ROLE_KEY = "user:%d:project:%d";
    public static final Duration USER_ROLE_TTL = Duration.ofMinutes(5);

    public static final String ROLE_PERMS_KEY = "role:%d:permissions";
    public static final Duration ROLE_PERMS_TTL = Duration.ofMinutes(10);

    public static final String INVITE_TOKEN_KEY = "invite:token:%s";
    public static final Duration INVITE_TOKEN_TTL = Duration.ofDays(1);

    public static final String ROLE_IS_OWNER_KEY = "role:%d:isOwner";
    public static final Duration ROLE_IS_OWNER_TTL = Duration.ofMinutes(10);

    private RedisConstants() {}
}

File: backend/sprints-service/Backend.Sprints.Api/Cache/RedisConstants.cs
namespace Backend.Sprints.Api.Cache
{
    public static class RedisConstants
    {
        public const string UserRoleKey = "user:{0}:project:{1}";
        public static readonly TimeSpan UserRoleTtl = TimeSpan.FromMinutes(5);

        public const string RolePermissionsKey = "role:{0}:permissions";
        public static readonly TimeSpan RolePermsTtl = TimeSpan.FromMinutes(10);

        public const string RoleIsOwnerKey = "role:{0}:isOwner";
        public static readonly TimeSpan RoleIsOwnerTtl = TimeSpan.FromMinutes(10);
    }

    public enum EntityType
    {
        PROJECT, ISSUE, SPRINT, COMMENT,
        ATTACHMENT, TAG, ANALYTICS, LOGS
    }

    public enum ActionType
    {
        VIEW, CREATE, EDIT, DELETE, ASSIGN, MANAGE,
        TAKE_ISSUE, CREATE_SUBTASK, SUBMIT_FOR_REVIEW,
        TRANSITION_CODE_REVIEW, TRANSITION_QA, FULL_TRANSITION,
        EDIT_OWN, DELETE_OWN, APPLY
    }
}
